### ISSUE TITLE
[BUG] Iterating forever

### ISSUE THREAD

**Issue by teamplayer3** [clementwanjau/tree-ds#16]
**Describe the bug**
Using the type `Node` as an iterator, iterates forever. 

**To Reproduce**
```
let tree = ...;
for node in tree.get_nodes() {
    // `node` is always first node
}
// not reachable
```

**Expected behavior**
Stop iterating when all nodes iterated.

**Solution**
In `impl Iterator::next()` you always create a new Iterator when calling next. This leads to the endless loop. It always returns the first node. You have to use a state in the iterator, which gets updated after every `next` call. In tests, you only check `nodes.iter()` but not `nodes.next()`.

**Comment by clementwanjau** [comment]
Thanks @teamplayer3 for bringing this up. 
This will be fixed in the upcoming release.

**Comment by clementwanjau** [comment]
This has been fixed in v0.1.6.
